packages/ui/src/services/color-config/plugin/create-color-config-plugin.ts

```diff
@@ -32,20 +33,17 @@ export const createColorConfigPlugin = (app: App) => {
     })
 
     colorNames.forEach((key) => {
-      setCSSVariable(`on-${key}`, getTextColor(newValue[key], newValue.textDark, newValue.textLight), root)
+      setCSSVariable(`on-${key}`, getTextColor(newValue[key], newValue.textDark, newValue.textLight, globalConfig.value.colors!.threshold), root)
     })
   }
 
-  updateColors(globalConfig.value.colors)
+  updateColors(globalConfig.value.colors!.variables)
 
   watch(() => globalConfig.value.colors, (newValue) => {
-    updateColors(newValue)
+    updateColors(newValue?.variables)
```

Comment:
Should we be looking at changes to the entire color configuration here, or are just the `variables` enough?
`watch(() => globalConfig.value.colors.variables, (newValue) => {`